### PR TITLE
feat(progress): Add interrupt option

### PR DESCRIPTION
**Pull request by jag3dagster**
Add the ability to specify if an action can be interrupted. Options that I've thought of off the bat are ped going ragdoll or ped being handcuffed.

### CODE REVIEW

File: resource/interface/client/progress.lua [hunk 1]
```diff
@@ -40,32 +40,48 @@ local function startProgress(data)
...
+					if IsPedRagdoll(cache.ped) then
+						progress = false
+					end
+				end
+
+				if data.interrupt.cuff then
+					if IsPedCuffed(cache.ped) then
+						progress = false
+					end
+				end
```
Comment (thelindat) [inline resource/interface/client/progress.lua:84]:
```lua
if data.interrupt.ragdoll and IsPedRagdoll(cache.ped) then
    progress = false
end

if data.interrupt.cuff and IsPedCuffed(cache.ped) then
    progress = false
end
```

As above would be better. These would likely also be better off as defaults (so you have to explicitly set them to false to disable), and should get checked before the progress bar begins.

**Review by thelindat — CHANGES_REQUESTED** [review]

File: resource/interface/client/progress.lua (hunk 1 above)
Comment (jag3dagster) [reply resource/interface/client/progress.lua]: Updated based on feedback

**Comment by jag3dagster** [comment]
Ope, realized I overlooked something... You can hold off on the review for a sec

**Comment by jag3dagster** [comment]
Alright, fixed what I overlooked and added falling to potential interrupts to ignore.

Interesting thing I realized during testing of this though was that ox_inventory useItem is already checking for dead / ragdoll / falling before attempting to use an item. Would y'all want that updated to be a check for ignoreInterrupt in client table defaulting to current behavior if no ignore is found?

**Pull request merged**